### PR TITLE
Lenses & Traversals

### PR DESCRIPTION
**Pull request by dialelo**
- [x] view/put/over
- [x] Implement plain lenses
- [x] Lens combinators
- [x] Implement traversals
- [x] Useful lenses/traversals
- [x] Documentation

### CODE REVIEW

**Comment by dialelo** [comment]
@niwinz @yurrriq take a look, it's ready!

**Comment by yurrriq** [comment]
Looks cool! With my limited experience using lenses, I don't feel qualified to comment on this one..

**Comment by arichiardi** [comment]
Very cool!

File: doc/content.adoc [hunk 1]
```diff
@@ -1825,6 +1825,317 @@ the monadic bind takes care of that. The type
...
+;; fst
+
+(l/focus l/fst [1 2 3])
+;; => 1
+
+(l/over l/fst inc [1 2 3])
+;; => [2 2 3]
+
+(l/put l/fst 42 [1 2 3])
+;; => [42 2 3]
```
Comment (niwinz) [inline doc/content.adoc]: Is this correct? I'm not clearly understand that behavior.

**Comment by niwinz** [comment]
Nice work! Feel free to merge it!

**Pull request merged**